components/api-gateway-migrator/pkg/finder/finder.go

```diff
@@ -0,0 +1,173 @@
+package finder
+
+import (
+	"fmt"
+
+	oldapiMeta "github.com/kyma-project/kyma/components/api-controller/pkg/apis/gateway.kyma-project.io/meta/v1"
+	oldapi "github.com/kyma-project/kyma/components/api-controller/pkg/apis/gateway.kyma-project.io/v1alpha2"
+	"github.com/kyma-project/kyma/components/api-gateway-migrator/pkg/migrator"
+	log "github.com/sirupsen/logrus"
+)
+
+type filter func(oldApi *oldapi.Api) (bool, string)
+
+var (
+	invalidStatusFilter = newStatusFilter()
+	labelFilter         = newLabelFilter
+	doubleJwtFilter     = newJwtStructureFilter()
+)
+
+type Finder struct {
+	filters   []filter
+	k8sClient *migrator.K8sClient
+}
+
+func (f *Finder) withFilters(filters []filter) *Finder {
+	f.filters = filters
+	return f
+}
+
+func (f *Finder) withClient(k8sClient *migrator.K8sClient) *Finder {
+	f.k8sClient = k8sClient
+	return f
+}
+
+func New(k8sClient *migrator.K8sClient, labels map[string]*string) *Finder {
+
+	return (&Finder{}).
+		withFilters([]filter{invalidStatusFilter, labelFilter(labels), doubleJwtFilter}).
+		withClient(k8sClient)
+}
+
+func (f *Finder) Find() ([]oldapi.Api, error) {
+
+	//List old api objects: apis.gateway.kyma-project.io/v1alpha2
+	oldApiList, err := f.k8sClient.FindOldApis()
+	if err != nil {
+		return nil, err
+	}
+
+	res := f.filterApis(oldApiList)
+
+	return res, nil
+}
+
+func (f *Finder) filterApis(oldApiList []oldapi.Api) []oldapi.Api {
+
+	res := []oldapi.Api{}
+
+	for _, oldApi := range oldApiList {
+
+		shouldSkip := false
+		for _, filter := range f.filters {
+			ol := oldApi
```

Comment:
Do we need this temp value?

Reply:
Yes, because you can't take an address of the loop variable. It's a known problem in golang: https://www.evanjones.ca/go-gotcha-loop-variables.html